examples/pytorch/dagnn/main.py

```diff
@@ -0,0 +1,272 @@
+import argparse
+from torch import nn
+from torch.nn import Parameter
+import dgl.function as fn
+from torch.nn import functional as F
+from dgl.data import CoraGraphDataset, CiteseerGraphDataset, PubmedGraphDataset
+import numpy as np
+import torch
+from tqdm import trange
+from utils import generate_random_seeds, set_random_state, evaluate
+
+
+class DAGNNConv(nn.Module):
+    def __init__(self,
+                 in_dim,
+                 k):
+        super(DAGNNConv, self).__init__()
+
+        self.s = Parameter(torch.FloatTensor(in_dim, 1))
+        self.k = k
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        gain = nn.init.calculate_gain('sigmoid')
+        nn.init.xavier_uniform_(self.s, gain=gain)
+
+    def forward(self, graph, feats):
+        with graph.local_scope():
+            results = [feats]
+
+            degs = graph.in_degrees().float().clamp(min=1)
+            norm = torch.pow(degs, -0.5)
+            norm = norm.to(feats.device).unsqueeze(1)
+
+            for _ in range(self.k):
+                feats = feats * norm
+                graph.ndata['h'] = feats
+                graph.update_all(fn.copy_u('h', 'm'),
+                             fn.sum('m', 'h'))
```

Comment:
Seems you need to fix the indent?

Reply:
Sorry, If we have added self-loop, it will be redundant.